src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -1910,5 +1916,89 @@ internal static float NumberToSingle(ref NumberBuffer number)
             float result = BitConverter.Int32BitsToSingle((int)(bits));
             return number.IsNegative ? -result : result;
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static bool IsHexPrefix(ReadOnlySpan<char> str, int i) =>
+            i + 1 < str.Length &&
+            str[i] == '0' &&
+            (str[i + 1] | 0x20) == 'x';
+
+        /// <summary>Tries to parse two characters with hex digits into <paramref name="currentByte"/>.</summary>
+        /// <remarks>Allows only hex digits (0-9, a-f, A-F).
+        /// If you want to parse plus sign and hex prefixes see <see cref="TryParseHexForTwoBytes(ReadOnlySpan{char}, int, ref byte, ref byte)"/>.</remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static bool TryParseHexStrict(ushort firstChar, ushort secondChar, ref byte currentByte)
+        {
+            const int maxLowBits = 15;
+            const int maxHighBits = 15 << 4;
+
+            byte[] charToHexLookupHi = s_charToHexLookupHi;
+            byte[] charToHexLookup = s_charToHexLookup;
+
+            unchecked
+            {
+
+                if (firstChar >= charToHexLookupHi.Length || secondChar >= charToHexLookup.Length)
+                    return false;
+
+                firstChar = charToHexLookupHi[firstChar];
```

Comment:
> I don't think this lookup table helps at all - multiplying by 16 is just one shift instruction, but this table takes up space in cache and local variables.

Thank you, fixed. Perhaps I made more than one change when I checked the performance of this option or was too lazy to run the benchmark several times.

> if ((firstChar | secondChar) == 0xFF) could do the check just as well.

Unfortunately, (0xFF | 0xFF) also equals to 0xFF.

I also tried such check - ```if ((firstChar|secondChar) >= 128)``` instead of ```if (firstChar >= charToHexLookupHi.Length || secondChar >= charToHexLookup.Length)```, but rollback it because of less code maintainability.

Reply:
The check for 0xFF must be before the shift, then it works nicely.
Shifting and adding can be done when returning the valid result.

Reply:
I am not sure about it, because for most cases (255 vs. 1) ```if (value >= byte.MaxValue)``` will be false. But anyway, [I'll try to rewrite parsing later](https://github.com/dotnet/coreclr/pull/21442#issuecomment-445486861) and this code will be removed.